### PR TITLE
ledger: move catchpoint sql code into storage package

### PR DESCRIPTION
**Pull request by icorderi**
## Summary

Moving some of the catchpoint sql queries into the store package. 

**Previous parts of this refactor:**
- part 1 - #4776

## Test Plan

Existing tests.

### CODE REVIEW

File: ledger/store/catchpoint.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+	"context"
+	"database/sql"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/util/db"
+)
+
+// CatchpointState is used to store catchpoint related variables into the catchpointstate table.
+type CatchpointState string
```
Comment (algorandskiy) [inline ledger/store/catchpoint.go:29]: // msgp:ignore CatchpointState

File: ledger/catchupaccessor.go [hunk 2]
```diff
@@ -179,16 +181,17 @@ type CatchupAccessorClientLedger interface {
 // MakeCatchpointCatchupAccessor creates a CatchpointCatchupAccessor given a ledger
 func MakeCatchpointCatchupAccessor(ledger *Ledger, log logging.Logger) CatchpointCatchupAccessor {
 	return &catchpointCatchupAccessorImpl{
-		ledger:        ledger,
-		stagingWriter: &stagingWriterImpl{wdb: ledger.trackerDB().Wdb},
-		log:           log,
+		ledger:          ledger,
+		catchpointStore: store.NewCatchpointSQLReaderWriter(ledger.trackerDB().Wdb.Handle),
```
Comment (algorandskiy) [inline ledger/catchupaccessor.go:185]: in theory we might use separate reader and writer handles. for sqlite it does not matter but still

File: ledger/catchpointtracker.go [hunk 3]
```diff
@@ -231,22 +237,23 @@ func (ct *catchpointTracker) finishFirstStage(ctx context.Context, dbRound basic
 	}
 
 	f := func(ctx context.Context, tx *sql.Tx) error {
+		cps := store.NewCatchpointSQLReaderWriter(tx)
```
Comment (algorandskiy) [inline ledger/catchpointtracker.go:240]: I'd say this requires a different abstraction like NewCatchpointSQLReaderWriter taking a handle and calling Atomic inside

Comment (icorderi) [reply ledger/catchpointtracker.go:240]: I'll rework transactions on a self contained PR after I finish moving the methods, they will require some extra abstractions and a fair bit of code rewrite.

**Review by algorandskiy — APPROVED** [review]

File: ledger/catchpointtracker.go (hunk 3 above)
Comment (algorandskiy) [reply ledger/catchpointtracker.go:240]: right, in subsequent ones

**Pull request merged**